nat-lab/bin/core-api.py

```diff
@@ -132,31 +186,89 @@ def add_node(self, req):
             public_key=req.public_key,
             os=req.os,
             os_version=req.os_version,
-            device_type=req.device_type,
-            nickname=req.nickname,
-            discovery_key=req.discovery_key,
-            relay_address=req.relay_address,
+            device_type=req.device_type or "",
+            nickname=req.nickname or "",
+            discovery_key=req.discovery_key or "",
+            relay_address=req.relay_address or "",
             traffic_routing_supported=req.traffic_routing_supported,
         )
 
         self.server.add_machine(node)
 
-        pprint(node)
         return node
 
     def handle_machines_delete(self):
         uid = self.path.removeprefix(self.machines_path)[1:]
-        print("uuid:", uid)
 
-        try:
-            self.server.remove_machine(uid)
-            print(f"{uid} removed")
-            self.send_response(204)
-        except KeyError:
-            print(f"{uid} unknown")
-            self.send_response(400)
+        if self.server.remove_machine(uid):
+            self._set_headers(status_code=HTTPStatus.NO_CONTENT)
+        else:
+            self._send_error_response(
+                CoreApiErrorCode.MACHINE_NOT_FOUND,
+                "Machine not found",
+                HTTPStatus.NOT_FOUND,
+            )
+
+    def handle_patch_machine(self, machine_id):
+        machine = self.server.get_machines().get(machine_id)
+        if not machine:
+            self._send_error_response(
+                CoreApiErrorCode.MACHINE_NOT_FOUND,
+                "Machine not found",
+                HTTPStatus.NOT_FOUND,
+            )
+            return
 
-        self.end_headers()
+        content_length = int(self.headers["Content-Length"])
```

Comment:
Note that I do not thing this header is required. And it is possible that it may be skipped. Although if both, esp32 and teliod includes it -> we may expect that it will exist here.